pragma-node/src/handlers/entries/create_perp_entry.rs

```diff
@@ -0,0 +1,192 @@
+use crate::config::config;
+use crate::infra::kafka;
+use crate::infra::repositories::publisher_repository;
+use crate::utils::{JsonExtractor, TypedData};
+use crate::AppState;
+use axum::extract::State;
+use axum::Json;
+use chrono::{DateTime, Utc};
+use pragma_entities::{EntryError, NewEntry, PublisherError};
+use serde::{Deserialize, Serialize};
+use starknet::core::crypto::{ecdsa_verify, Signature};
+use starknet::core::types::FieldElement;
+
+use super::{CreatePerpEntryRequest, CreatePerpEntryResponse, PerpEntry};
+
+#[derive(Debug, Serialize, Deserialize)]
+struct PublishMessage {
+    action: String,
+    perp_entries: Vec<PerpEntry>,
+}
+
+// TODO(akhercha): double-check this
+fn build_publish_message(
+    perp_entries: &[PerpEntry],
+) -> Result<TypedData<PublishMessage>, EntryError> {
+    // Construct the raw string with placeholders for the entries
+    let raw_message = format!(
+        r#"{{
+            "domain": {{"name": "Pragma", "version": "1"}},
+            "primaryType": "Request",
+            "message": {{
+                "action": "Publish",
+                "perp_entries": {}
+            }},
+            "types": {{
+                "StarkNetDomain": [
+                    {{"name": "name", "type": "felt"}},
+                    {{"name": "version", "type": "felt"}}
+                ],
+                "Request": [
+                    {{"name": "action", "type": "felt"}},
+                    {{"name": "perp_entries", "type": "PerpPentry*"}}
+                ],
+                "PerpEntry": [
```

Comment:
so this message has to be exactly the same as the one constructed and signed in python. So in python/cairo we do not have such thing as `PerpEntry`.  But rather `FutureEntry` with an expiration timestamp. I think we should align on this implementation here as well.

Reply:
done here:
https://github.com/astraly-labs/pragma-sdk/pull/109